docs/vfio-user.rst

```diff
@@ -517,7 +549,11 @@ Reply
 ^^^^^
 
 The same message format is used in the server's reply with the semantics
-described above.
+described above. In case the client set ``twin_socket`` to true in its
+capabilities, the server may include a file descriptor to use for the
+server-to-client command channel in the reply. The index of the file descriptor
+in the ancillary data of the reply is given by the ``twin_socket`` capability
```

Comment:
is it? It's a boolean above.

Reply:
Good point. I had been reusing the name with a different type so it's clear this is about the same thing, but  `twin_socket_fd_index` will likely be less confusing. Let me know if you disagree.

Reply:
I think twin_socket on the outgoing (request) and twin_socket_fd_index on the reply ??

Reply:
actually, I think maybe have two: twin_socket both ways, and an *additional* twin_socket_fd_index in the reply

Reply:
What would be twin_socket's purpose in the reply? Do you have a scenario in mind where we'd want to encode twin_socket = false, but still provide a twin_socket_fd_index value?

If you're coming from a symmetry of feature presence point of view, then I would suggest we introduce `twin_socket` as a dictionary, carrying an (optional) `enable` boolean in the client message, and an optional `fd_index` in the server reply.

Reply:
dict sounds good